tools/serverpod_cli/bin/util/prompt_to_update.dart

```diff
@@ -0,0 +1,131 @@
+
+import 'print.dart';
+
+/// Check current Serverpod CLI version and prompt user to update if needed
+Future<void> promptToUpdateIfNeeded() async {
+  try {
+    String? currentVersion = _getPackageVersion();
+    String? packageName = _getPackageName();
+    if (currentVersion == null || packageName == null) return;
+    _Package? package = await _PackageService.getLatestPackage(packageName);
+    if (package == null) return;
+    bool isUpToDate = package.version == currentVersion;
+
+    if (!isUpToDate) {
+      print('');
+      print(_promptMessage(package.version));
+      print('');
+    }
+  } catch (e) {
+    print(e);
+  }
+}
+
+/// Update Serverpod CLI to the latest available version
+Future<void> updateCLI({bool verbose = false}) async {
+  try {
+    String? currentVersion = _getPackageVersion();
+    String? packageName = _getPackageName();
+    if (currentVersion == null || packageName == null) return;
+    _Package? package = await _PackageService.getLatestPackage(packageName);
+    if (package == null) return;
+    if (package.version == currentVersion) {
+      return printww('Serverpod is up to date 😉');
+    }
+    ProcessResult upgrade = await Process.run(
+      'dart',
+      [
+        'pub',
+        'global',
+        'activate',
+        'serverpod',
+      ],
+    );
+    if (upgrade.exitCode == 0) {
+      stdout.write(upgrade.stdout);
+      stderr.write(upgrade.stderr);
+      printwwln('Serverpod CLI upgraded successfully');
+      return;
+    } else {
+      if (verbose) printwwln(upgrade.stderr.toString());
+      printwwln('Something went wrong, Failed to upgrade Serverpod CLI.');
+      return;
+    }
+  } on Exception catch (e) {
+    if (verbose) printwwln(e.toString());
+    printwwln('Something went wrong, Failed to upgrade Serverpod CLI.');
+    return;
+  }
+}
+
+YamlMap? _getcurrentPackage() {
+  File file = File('pubspec.yaml');
+  String str = file.readAsStringSync();
+  YamlDocument doc = loadYamlDocument(str);
+  YamlNode contents = doc.contents;
+  if (contents is! YamlMap) return null;
+  return contents;
+}
+
+String? _getPackageVersion() {
+  YamlMap? package = _getcurrentPackage();
+
+  return package?.nodes['version']?.value.toString();
+}
+
+String? _getPackageName() {
+  YamlMap? package = _getcurrentPackage();
+
+  return package?.nodes['name']?.value.toString();
+}
+
+abstract class _PackageService {
+  static Future<_Package?> getLatestPackage(String name) async {
+    try {
+      Response respose =
+          await get(Uri.parse('https://pub.dev/api/packages/$name'));
+      Map<String, dynamic> map = jsonDecode(respose.body);
+      return _Package(name: map['name'], version: map['latest']['version']);
+    } catch (e) {
+      print(e);
+      return null;
+    }
+  }
+}
+
+class _Package {
+  final String name;
+  final String version;
+  _Package({
+    required this.name,
+    required this.version,
+  });
+}
+
+String _promptMessage(String version) {
```

Comment:
This feels a bit like a hack. I would either use that other package or write this in a more reusable way.